streams/cluster/src/main/java/com/hortonworks/streamline/streams/cluster/discovery/ambari/AmbariServiceNodeDiscoverer.java

```diff
@@ -181,9 +183,29 @@ public void init(Map<String, String> conf) throws ConfigException {
     }
   }
 
+  private List<String> createAmbariConfNameList(ServiceConfigurations serviceConfigurations) {
+    String[] confNames = serviceConfigurations.getConfNames();
+    return Arrays.stream(confNames).map(confName -> {
+      ServiceConfigTypeAmbariToOriginalPattern pattern = ServiceConfigTypeAmbariToOriginalPattern.findByOriginalConfType(confName);
+      if (pattern != null) {
+        return pattern.ambariConfType();
+      }
+      return confName;
+    }).collect(toList());
+  }
+
+  private String getOriginConfigTypeName(String confItemType) {
+    ServiceConfigTypeAmbariToOriginalPattern pattern = ServiceConfigTypeAmbariToOriginalPattern.findByAmbariConfType(confItemType);
+    if (pattern != null) {
+      return pattern.originConfType();
+    }
+    return confItemType;
```

Comment:
I think it should not happen, but we could apply defensive programming: place precondition check here, and I'm in favor of doing this. Will fix.